rasa/core/channels/facebook.py

```diff
@@ -141,7 +143,21 @@ def send(self, recipient_id: Text, element: Any) -> None:
         # this is a bit hacky, but the client doesn't have a proper API to
         # send messages but instead expects the incoming sender to be present
         # which we don't have as it is stored in the input channel.
-        self.messenger_client.send(element.to_dict(), recipient_id, "RESPONSE")
+        
+        elementlenght = element.to_dict()
+       
+        if 'text' in elementlenght:
+            messagelength = len(elementlenght['text'])
+            delay = messagelength * fb_delay
+            typing_on = SenderAction(sender_action='typing_on')
+            self.messenger_client.send_action(typing_on.to_dict(), recipient_id)
+            time.sleep(delay)
```

Comment:
change to `time.sleep(delay)` to `asyncio.sleep(delay)` to prevent negative interaction with sanic.

Reply:
This change necessitates changing the send function to an async function and adding `await` to all it's invocations.

Reply:
Hi Will,

your changes were applied to this PR